processor/src/main/java/org/mapstruct/ap/internal/model/BeanMappingMethod.java

```diff
@@ -393,26 +393,35 @@ private boolean handleDefinedMappings() {
             }
 
             for ( Map.Entry<String, List<Mapping>> entry : methodMappings.entrySet() ) {
+                String target = null;
```

Comment:
Any reasons you define `target` in this scope and not where it is used? (line 400ff)

Reply:
oops.. that's a remanence of a previous iteration.. Spent a lot of time on this one.. (put System.out.println's to find out how this works exactly.. I'll fix this.